### PR TITLE
Combined Fixes from PR #644, #640, and #618

### PR DESCRIPTION
**Pull request by abdelaziz-mahdy**
This pull request combines changes from several individual pull requests to avoid merge conflicts. The included pull requests are as follows:

- PR [#644](https://github.com/media-kit/media-kit/pull/644)
- PR [#640](https://github.com/media-kit/media-kit/pull/640)
- PR [#618](https://github.com/media-kit/media-kit/pull/618)

feel free to merge this incase all is good

summary 

**Overview:**
This pull request introduces a series of enhancements and bug fixes to the video player controls, improving the overall user experience and responsiveness. Key areas of focus include widget width handling, gesture detection, and the double-tap seek UI, along with a new feature for mouse visibility control.

**Fixes and Enhancements:**

1. **Widget Width Handling Issue:**
   - **Problem:** Inaccurate control scaling due to using MediaQuery for deriving control widths, causing issues in non-full-screen modes.
   - **Fix:** Implemented LayoutBuilder for dynamic width acquisition, ensuring correct sizing and positioning of controls in various player sizes.

2. **Gesture Detection Delay:**
   - **Problem:** Delay in gesture detection impacting immediate control responsiveness.
   - **Fix:** Segmented logic and immediate touch event listener akin to YouTube's approach for faster control display on touch.

3. **Double Tap Seek UI Adjustment:**
   - **Problem:** Overlapping of double-tap-to-seek function with normal interaction areas.
   - **Fix:** Segmented layout (2,1,2 ratio) with double-tap-to-seek on sides and regular taps in the center, mirroring YouTube's UI for clear differentiation and intuitive use. can be edited by developers

4. **Seek Button Visibility Issue:**
   - **Problem:** Seek buttons hidden yet clickable due to conflicting _hideSeekForwardButton and _mountSeekForwardButton states.
   - **Fix:** implemented animated opacity with timers for consistent behavior.

**New Feature:**

- **Hide Mouse on Controls Removal:**
  - **Addition:** New `hideMouseOnControlsRemoval` property in `MaterialDesktopVideoControlsThemeData`.
  - **Functionality:** Hides mouse cursor when video controls are removed, offering a cleaner full-screen video experience.
  - **Default Behavior:** Set to false to maintain backward compatibility.
  - **Windows:** https://github.com/media-kit/media-kit/pull/618#issuecomment-1826046719
**Usage Example:**

```dart
MaterialDesktopVideoControlsThemeData(
  // ... other properties ...
  hideMouseOnControlsRemoval: true,
)
```

This pull request collectively addresses various aspects of video player control usability, enhancing both the aesthetic and functional experience for end-users.

### CODE REVIEW

**Comment by user1121114685** [comment]
Great update, it happens to be related to double-clicking. Can I take the liberty of asking a question? I want to customize the double-click function and use double-click to pause playback. Can I only modify the source code?

**Comment by zezo357** [comment]
> Great update, it happens to be related to double-clicking. Can I take the liberty of asking a question? I want to customize the double-click function and use double-click to pause playback. Can I only modify the source code?

sadly for this you will need to implement your own controls

**Comment by user1121114685** [comment]
> > Great update, it happens to be related to double-clicking. Can I take the liberty of asking a question? I want to customize the double-click function and use double-click to pause playback. Can I only modify the source code?
> 
> sadly for this you will need to implement your own controls

thank you for your reply,this is a good idea

**Comment by abdelaziz-mahdy** [comment]
since i am using this branch on my own app i added the tap to pause and play using the https://github.com/media-kit/media-kit/pull/742 which is showing in the commits 

and the padding under `bottom: 16.0 + subtitleVerticalShiftOffset,` to be able to use the slider since it was overlayed

File: media_kit_video/lib/media_kit_video_controls/src/controls/material.dart [hunk 1]
```diff
@@ -461,7 +488,7 @@ class _MaterialVideoControls extends StatefulWidget {
 class _MaterialVideoControlsState extends State<_MaterialVideoControls> {
   late bool mount = _theme(context).visibleOnMount;
   late bool visible = _theme(context).visibleOnMount;
-
+  late double _widgetWidth;
```
Comment (moffatman) [inline media_kit_video/lib/media_kit_video_controls/src/controls/material.dart:491]: This is a bad approach. Other 2 better ways

1. Just do it all in widget tree. Put in a stack above the video or whatever, three different gesture detectors in a row with different callbacks

2. Stay with one gesture detector. But get the rect on-tap using
```dart
(context.findRenderObject() as RenderBox).paintBounds
```

Comment (abdelaziz-mahdy) [reply media_kit_video/lib/media_kit_video_controls/src/controls/material.dart]: i prefer the 2 approach and implemented it

**Review by moffatman — APPROVED** [review]
LGTM

**Pull request merged**